features/google/src/model/importers/utils/lang-constants.js

```diff
@@ -0,0 +1,31 @@
+export const langConstants = {
+    AccessLogImporter: {
+        english: { regex: /\/Access Log Activity\/.*\.csv$/ },
+        german: { regex: /\/Zugriffsprotokollaktivitäten\/.*\.csv$/ },
+        danish: { regex: /\/Aktiviteter i adgangsloggen\/.*\.csv$/ },
+    },
+    ActivitiesImporter: {
+        english: { regex: /\/My Activity\/.*\.html$/ },
+        german: { regex: /\/Meine Aktivitäten\/.*\.html$/ },
+        danish: { regex: /\/Mine aktiviteter\/.*\.html$/ },
+    },
+    SemanticLocationsImporter: {
+        english: {
+            regex: /\/[^/]+\/Semantic Location History\/\d+\/[^.]+\.json$/,
+        },
+        german: {
+            regex: /\/[^/]+\/Semantic Location History\/\d+\/[^.]+\.json$/,
+        },
+        danish: {
+            regex: /\/[^/]+\/Semantic Location History\/\d+\/[^.]+\.json$/,
+        },
+    },
+};
+export function matchRegex(path, importer) {
```

Comment:
Also, maybe consider using `importer = matchRegex.caller` by default? That would give you the function or class that's calling (some parsing might be needed, additionally)

Reply:
that's even smoother but could it not get a bit confusing ?

Reply:
Confusing in which sense? It's also simply a default, you can override it to make it explicit. On the other hand, if the value is available by introspection, why would we want to set it explicitly?

Reply:
true true

Reply:
it just feels a bit weird to be pulling up the stack in order to figure out the importer's id. Passing down "this" feels a bit more explicit and forward. so I'd lean on the former method on acquiring the importer's ID

Reply:
We do that in Perl all the time. Raku, too...